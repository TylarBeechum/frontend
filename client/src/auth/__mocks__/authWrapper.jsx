/* eslint-disable */
export const mock = jest.fn();

export const useAuth0 = () => {
  return {
    loginWithRedirect: mock
  };
};
