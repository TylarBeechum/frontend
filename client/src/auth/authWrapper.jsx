import React, { useState, useEffect, useContext } from 'react';
import createAuth0Client from '@auth0/auth0-spa-js';
import { useStateValue } from 'react-conflux';
import { globalContext } from '../store/reducers/globalReducer';
import { register } from './authQueries';

const DEFAULT_REDIRECT_CALLBACK = () =>
  window.history.replaceState({}, document.title, window.location.pathname);

export const Auth0Context = React.createContext();
export const useAuth0 = () => useContext(Auth0Context);
export const Auth0Provider = ({
  history,
  children,
  onRedirectCallback = DEFAULT_REDIRECT_CALLBACK,
  ...initOptions
}) => {
  const [, dispatchGlobal] = useStateValue(globalContext);
  const [isAuthenticated, setIsAuthenticated] = useState();
  const [user, setUser] = useState();
  const [auth0Client, setAuth0] = useState();
  const [loading, setLoading] = useState(true);
  const [popupOpen, setPopupOpen] = useState(false);

  useEffect(() => {
    const initAuth0 = async () => {
      const auth0FromHook = await createAuth0Client(initOptions);
      setAuth0(auth0FromHook);

      if (window.location.search.includes('code=')) {
        const { appState } = await auth0FromHook.handleRedirectCallback();
        onRedirectCallback(appState);
      }

      const isAuthenticated = await auth0FromHook.isAuthenticated();

      setIsAuthenticated(isAuthenticated);

      if (isAuthenticated) {
        const user = await auth0FromHook.getUser();
        setUser(user);

        const authToken = await auth0FromHook.getTokenSilently({
          audience: `${process.env.REACT_APP_AUTH0_AUDIENCE}`,
          scope: 'openid profile email offline_access'
        });
        console.log(authToken);
        
        const userWithAuth0 = {
          email: user.email,
          picture: user.picture,
          authToken
        };

        try {
          const result = await register(userWithAuth0);
          dispatchGlobal({ type: 'REGISTER', payload: result.data.addUser });

          localStorage.id = result.data.addUser.id;

          if (result.data.addUser.roleId === null) {
            history.push('/onboarding');
          } else {
            history.push('/dashboard');
          }
        } catch (error) {
          console.log(error);
        }
      }

      setLoading(false);
    };
    initAuth0();
    // eslint-disable-next-line
  }, []);

  const loginWithPopup = async (params = {}) => {
    setPopupOpen(true);
    try {
      await auth0Client.loginWithPopup(params);
    } catch (error) {
      console.error(error);
    } finally {
      setPopupOpen(false);
    }
    const user = await auth0Client.getUser();
    setUser(user);
    setIsAuthenticated(true);
  };

  const handleRedirectCallback = async () => {
    setLoading(true);
    await auth0Client.handleRedirectCallback();
    const user = await auth0Client.getUser();
    setLoading(false);
    setIsAuthenticated(true);
    setUser(user);
  };
  return (
    <Auth0Context.Provider
      value={{
        isAuthenticated,
        user,
        loading,
        popupOpen,
        loginWithPopup,
        handleRedirectCallback,
        getIdTokenClaims: (...p) => auth0Client.getIdTokenClaims(...p),
        loginWithRedirect: (...p) => auth0Client.loginWithRedirect(...p),
        getTokenSilently: (...p) => auth0Client.getTokenSilently(...p),
        getTokenWithPopup: (...p) => auth0Client.getTokenWithPopup(...p),
        logout: (...p) => auth0Client.logout(...p)
      }}
    >
      {children}
    </Auth0Context.Provider>
  );
};
