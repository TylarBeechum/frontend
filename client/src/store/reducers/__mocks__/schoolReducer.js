import { createContext } from 'react';

export const SCHOOL_DATA_START = 'SCHOOL_DATA_START';
export const SCHOOL_DATA_SUCCESS = 'SCHOOL_DATA_SUCCESS';
export const SCHOOL_DATA_ERROR = 'SCHOOL_DATA_ERROR';
export const SEARCH_HANDLE_CHANGE = 'SEARCH_HANDLE_CHANGE';
export const REMOVE_CREDENTIAL_START = 'REMOVE_CREDENTIAL_START';
export const REMOVE_CREDENTIAL_SUCCESS = 'REMOVE_CREDENTIAL_SUCCESS';
export const REMOVE_CREDENTIAL_ERROR = 'REMOVE_CREDENTIAL_ERROR';
export const UPDATE_CRED_DATA = 'UPDATE_CRED_DATA';

export const schoolContext = createContext();

const initialState = {
  schoolData: {
    schoolDetails: {
      name: 'testName',
      credentials: [
        {
          id: 1,
          credName: 'test1',
          criteria: 'test',
          ownerName: 'Nathan Thomas'
        },
        {
          id: 2,
          credName: 'test2',
          criteria: 'test',
          ownerName: 'Nathan Thomas'
        },
        {
          id: 3,
          credName: 'test3',
          criteria: 'test',
          ownerName: 'Nathan Thomas'
        }
      ]
    }
  },
  schoolDataSuccess: true,
  schoolDataStart: false,
  schoolDataError: false,
  schoolSearchInput: '',
  removeCredentialStart: false,
  removeCredentialSuccess: false,
  removeCredentialError: false
};

export const schoolReducer = (state = initialState, action) => {
  switch (action.type) {
    case SCHOOL_DATA_START:
      return {
        ...state,
        schoolDataStart: true,
        schoolDataSuccess: false,
        schoolDataError: false
      };
    case SCHOOL_DATA_SUCCESS:
      return {
        ...state,
        schoolDataStart: false,
        schoolDataSuccess: true,
        schoolDataError: false,
        schoolData: action.payload.data.getUserById
      };
    case SCHOOL_DATA_ERROR:
      return {
        ...state,
        schoolDataError: true,
        schoolDataStart: false
      };
    case SEARCH_HANDLE_CHANGE:
      return {
        ...state,
        schoolSearchInput: action.payload
      };
    case REMOVE_CREDENTIAL_START:
      return {
        ...state,
        removeCredentialStart: true,
        removeCredentialSuccess: false,
        removeCredentialError: false
      };
    case REMOVE_CREDENTIAL_SUCCESS:
      return {
        ...state,
        removeCredentialStart: false,
        removeCredentialSuccess: true,
        removeCredentialError: false,
        schoolData: {
          ...state.schoolData,
          schoolDetails: {
            ...state.schoolData.schoolDetails,
            credentials: state.schoolData.schoolDetails.credentials.filter(
              credential => credential.id !== action.payload.credId
            )
          }
        }
      };
    case REMOVE_CREDENTIAL_ERROR:
      return {
        ...state,
        removeCredentialStart: false,
        removeCredentialSuccess: false,
        removeCredentialError: true
      };
    case UPDATE_CRED_DATA:
      return {
        ...state,
        schoolData: {
          ...state.schoolData,
          schoolDetails: {
            ...state.schoolData.schoolDetails,
            credentials: action.payload
          }
        }
      };
    default:
      return state;
  }
};
