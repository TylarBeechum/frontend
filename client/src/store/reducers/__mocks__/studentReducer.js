import { createContext } from 'react';

export const STUDENT_DATA_START = 'STUDENT_DATA_START';
export const STUDENT_DATA_SUCCESS = 'STUDENT_DATA_SUCCESS';
export const STUDENT_DATA_ERROR = 'STUDENT_DATA_ERROR';
export const SEARCH_HANDLE_CHANGE = 'SEARCH_HANDLE_CHANGE';
export const SET_STUDENT_DATA = 'SET_STUDENT_DATA';
export const STUDENT_EMAIL_UPDATE = 'STUDENT_EMAIL_UPDATE';
export const REMOVE_STUDENT_EMAIL = 'REMOVE_STUDENT_EMAIL';

export const studentContext = createContext();

const initialState = {
  studentData: {
    fullName: 'test',
    studentDetails: {
      credentials: [
        {
          id: 1,
          credName: 'test1',
          criteria: 'test',
          ownerName: 'Nathan Thomas'
        },
        {
          id: 2,
          credName: 'test2',
          criteria: 'test',
          ownerName: 'Nathan Thomas'
        },
        {
          id: 3,
          credName: 'test3',
          criteria: 'test',
          ownerName: 'Nathan Thomas'
        }
      ]
    }
  },
  studentDataSuccess: true,
  studentDataStart: false,
  studentDataError: false,
  studentSearchInput: ''
};

export const studentReducer = (state = initialState, action) => {
  switch (action.type) {
    case STUDENT_DATA_START:
      return {
        ...state,
        studentDataStart: true,
        studentDataSuccess: false,
        studentDataError: false
      };
    case STUDENT_DATA_SUCCESS:
      return {
        ...state,
        studentDataStart: false,
        studentDataSuccess: true,
        studentDataError: false,
        studentData: action.payload.data.getUserById
      };
    case STUDENT_DATA_ERROR:
      return {
        ...state,
        studentDataError: true,
        studentDataStart: false
      };
    case SEARCH_HANDLE_CHANGE:
      return {
        ...state,
        studentSearchInput: action.payload
      };
    case SET_STUDENT_DATA:
      return {
        ...state,
        studentData: {
          ...state.studentData,
          studentDetails: action.payload.data.addStudentDetail
        },
        studentDataSuccess: true
      };
    case STUDENT_EMAIL_UPDATE:
      return {
        ...state,
        studentData: {
          ...state.studentData,
          studentDetails: {
            ...state.studentData.studentDetails,
            emailList: [
              ...state.studentData.studentDetails.emailList,
              action.payload
            ]
          }
        }
      };
    case REMOVE_STUDENT_EMAIL:
      return {
        ...state,
        studentData: {
          ...state.studentData,
          studentDetails: {
            ...state.studentData.studentDetails,
            emailList: action.payload
          }
        }
      };
    default:
      return state;
  }
};
