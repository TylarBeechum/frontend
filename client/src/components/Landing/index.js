import React from 'react';
import styled from 'styled-components';
import { useAuth0 } from '../../auth/authWrapper';
import PropTypes from 'prop-types';
import { BaseButton } from '../../styles/themes';

function Landing({ history }) {
  const { loginWithRedirect } = useAuth0();
  return (
    <LandingPage>
      <BaseButton
        type="button"
        onClick={() => loginWithRedirect({})}
        label="Get Started"
      />
    </LandingPage>
  );
}

const LandingPage = styled.div`
  background-color: #3aecfc;
  min-height: 91vh;
`;

Landing.propTypes = {
  history: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

export default Landing;
