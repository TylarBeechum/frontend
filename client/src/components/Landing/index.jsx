import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { useAuth0 } from '../../auth/authWrapper';
import { BaseButton } from '../../styles/themes';
import stampdBackground from '../../images/stampdbackground.svg';

function Landing({ history }) {
  const { loginWithRedirect } = useAuth0();
  return (
    <LandingPage>
      <div className="landing-img">
        <img src={stampdBackground} alt="" draggable="false" />
      </div>
      <div className="cta-container">
        <div className="cta-text">
          Lorem Epsom in the blockchain with stampd
        </div>
        <BaseButton
          type="button"
          onClick={() => loginWithRedirect({})}
          label="Get Started"
          primary
        />
      </div>
    </LandingPage>
  );
}

const LandingPage = styled.div`
  height: 100vh;
  width: 100%;
  overflow: hidden;
  position: relative;

  .landing-img {
    transform: translateX(-35%) translateY(-18%) scale(0.8);
    position: absolute;
    left: 0;
    z-index: -1;
    overflow: hidden;
  }

  .cta-container {
    display: flex;
    height: 100%;
    flex-direction: column;
    justify-content: center;
    align-items: flex-end;
    .cta-text {
      font-size: 2.8rem;
    }
  }
`;

Landing.propTypes = {
  history: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

export default Landing;
