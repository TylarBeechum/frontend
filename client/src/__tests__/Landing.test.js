/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import {
  renderWithRouterAndProviders,
  rendererWithRouterAndProviders
} from '../__testHelpers__';

import Landing from '../components/Landing/index';
import { mock } from '../auth/__mocks__/authWrapper';

describe('<Landing />', () => {
  describe('should render the component successfully without crashing', () => {
    it('tests that the component does not crash when it is rendered', () => {
      const helpers = renderWithRouterAndProviders(<Landing />);
    });
  });

  describe('should match the previous snapshot of the component', () => {
    it('tests that the component has not changed and matched the previous snapshot', () => {
      const tree = rendererWithRouterAndProviders(<Landing />);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe('should render the marketing text to the screen successfully', () => {
    it('tests that the 1/3 string on the landing page is rendered successfully', () => {
      const { getByText } = renderWithRouterAndProviders(<Landing />);
      const string = getByText(/Easy. Verified. Blockchain./i);
      expect(string).toBeDefined();
    });

    it('tests that the 2/3 string on the landing page is rendered successfully', () => {
      const { getByText } = renderWithRouterAndProviders(<Landing />);
      const string = getByText(/Issue credentials with/i);
      expect(string).toBeDefined();
    });

    it('tests that the 3/3 string on the landing page is rendered successfully', () => {
      const { getByText } = renderWithRouterAndProviders(<Landing />);
      const string = getByText(/Stampd/i);
      expect(string).toBeDefined();
    });
  });

  describe('should render the landing page button to the screen successfully', () => {
    it('tests that the "Get Started" button is rendered successfully', () => {
      const { getByText } = renderWithRouterAndProviders(<Landing />);
      const getStartedBtn = getByText(/Get Started/i);
      expect(getStartedBtn).toBeDefined();
    });
  });
});
