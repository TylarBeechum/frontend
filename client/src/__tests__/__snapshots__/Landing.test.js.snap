// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 iDwiuh"
>
  <main
    className="sc-EHOje cTUAzB"
  >
    <div>
      <div
        className="landing-img"
      >
        <img
          alt=""
          draggable="false"
          src="stampdbackground.svg"
        />
      </div>
      <div
        className="cta-container"
      >
        <div
          className="cta-text"
        >
          <h1>
            Easy. Verified. Blockchain.
            <br />
            Issue credentials with
            <span>
               Stampd
            </span>
          </h1>
        </div>
        <button
          className="sc-bdVaJa bDWFJH StyledButton-sc-323bzc-0 aNlbA"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          size="xlarge"
          type="button"
        >
          Get Started
        </button>
      </div>
    </div>
  </main>
</div>
`;
